kymatio/scattering1d/frontend/base_frontend.py

```diff
@@ -0,0 +1,258 @@
+            or collected into a dictionary. For more details, see the
+            documentation for `forward()`.
+        """
+    def __init__(self, J, shape, Q=1, max_order=2, average=True, oversampling=0, vectorize=True, backend=None):
+        super(ScatteringBase1D, self).__init__()
+        self.J = J
+        self.shape = shape
+        self.Q = Q
+        self.max_order = max_order
+        self.average = average
+        self.oversampling = oversampling
+        self.vectorize = vectorize
+        self.backend = backend
+
+    def build(self):
+        """Set up padding and filters
+
+        Certain internal data, such as the amount of padding and the wavelet
+        filters to be used in the scattering transform, need to be computed
+        from the parameters given during construction. This function is called
+        automatically during object creation and no subsequent calls are
+        therefore needed.
+        """
+        self.r_psi = math.sqrt(0.5)
+        self.sigma0 = 0.1
+        self.alpha = 5.
+        self.P_max = 5
+        self.eps = 1e-7
+        self.criterion_amplitude = 1e-3
+        self.normalize = 'l1'
+
+        # check the shape
+        if isinstance(self.shape, numbers.Integral):
+            self.T = self.shape
+        elif isinstance(self.shape, tuple):
+            self.T = self.shape[0]
+            if len(self.shape) > 1:
+                raise ValueError("If shape is specified as a tuple, it must "
+                                 "have exactly one element")
+        else:
+            raise ValueError("shape must be an integer or a 1-tuple")
+
+        # Compute the minimum support to pad (ideally)
+        min_to_pad = compute_minimum_support_to_pad(
+            self.T, self.J, self.Q, r_psi=self.r_psi, sigma0=self.sigma0,
+            alpha=self.alpha, P_max=self.P_max, eps=self.eps,
+            criterion_amplitude=self.criterion_amplitude,
+            normalize=self.normalize)
+        # to avoid padding more than T - 1 on the left and on the right,
+        # since otherwise torch sends nans
+        J_max_support = int(np.floor(np.log2(3 * self.T - 2)))
+        self.J_pad = min(int(np.ceil(np.log2(self.T + 2 * min_to_pad))),
+                         J_max_support)
+        # compute the padding quantities:
+        self.pad_left, self.pad_right = compute_padding(self.J_pad, self.T)
+        # compute start and end indices
+        self.ind_start, self.ind_end = compute_border_indices(
+            self.J, self.pad_left, self.pad_left + self.T)
+
+    def create_filters(self):
+        # Create the filters
+        self.phi_f, self.psi1_f, self.psi2_f, _ = scattering_filter_factory(
+            self.J_pad, self.J, self.Q, normalize=self.normalize,
+            criterion_amplitude=self.criterion_amplitude,
+            r_psi=self.r_psi, sigma0=self.sigma0, alpha=self.alpha,
+            P_max=self.P_max, eps=self.eps)
+
+    def meta(self):
```

Comment:
Do we want this in the other scattering transforms too?

Reply:
yes, in the base

Reply:
Maybe we want a base meta scattering class, just to separate these things out into logical units? 

@janden @lostanlen thoughts?